### ISSUE TITLE
Dispatch: Special Issues checkboxes are not grouped together (End: 7/10)

### ISSUE THREAD

**Issue by alisan16** [department-of-veterans-affairs/caseflow#12123]
Checkboxes need to be grouped and labeled for 'Special Issues' on Establish Next Claim page

On the Establish Next claim page the label: Special Issue(s) is not read with the check box selections for assistive technology users. (New: 08/27/2019) The boxes need to be grouped and the label read “select special issue(s)” with all the checkboxes

Suggested fix: https://www.w3.org/WAI/tutorials/forms/grouping/

**Comment by ajspotts** [comment]
Alisa can you tell me where this view (the url) is located?

**Comment by alisan16** [comment]
Sure - these issues are all related to Caseflow Demo: www.caseflowdemo.com

This specific issue is on http://www.caseflowdemo.com/dispatch/establish-claim/1#/decision
You can test it out with test user: BVAAABSHIRE (VACO) on Caseflow Demo and then navigating to Dispatch > Work History > Establish Next Claim then scroll down the page until you see the checkboxes.

<img width="1054" alt="Screen Shot 2019-10-18 at 3 23 15 PM" src="https://user-images.githubusercontent.com/1198106/67122213-4c223780-f1bb-11e9-8651-26f712f2a764.png">

<img width="1048" alt="Screen Shot 2019-10-18 at 3 24 07 PM" src="https://user-images.githubusercontent.com/1198106/67122229-580df980-f1bb-11e9-885e-f0d25ed5f381.png">

<img width="1059" alt="Screen Shot 2019-10-18 at 3 24 30 PM" src="https://user-images.githubusercontent.com/1198106/67122297-7f64c680-f1bb-11e9-811d-54862274f9d6.png">

**Comment by ajspotts** [comment]
@alisan16 Does a solution to this problem require testing with JAWS or is VoiceOver testing considered to be sufficient? Asking after I peaked in on [issue 12096](https://github.com/department-of-veterans-affairs/caseflow/issues/12096)

**Comment by alisan16** [comment]
@ajspotts Yes, please take a look at the linked epic for instructions for how to reproduce and test the issues. You can use the standard screenreader installed on your computer, which on Mac is Voiceover and on Windows is Narrator, I believe. If you have access to a GFE or CAG Windows computer, the JAWS program should be installed and you can use that as well.

https://github.com/department-of-veterans-affairs/caseflow/issues/12095

**Comment by ajspotts** [comment]
Resolved by this [ticket](https://github.com/department-of-veterans-affairs/caseflow/pull/12559)

**Comment by araposo-tistatech** [comment]
Reopening, per the latest accessibility audit this is not resolved: 

* The following occurs with the Dispatch application:
    * On the Establish Next claim page the label: Special Issue(s) is not read with the check box selections for assistive technology users. (New: 08/27/2019) The boxes need to be grouped and the label read “select special issue(s)” with all the checkboxes (Not Resolved 3/26/2020)

**Comment by leikkisa** [comment]
@araposo-tistatech 
I tested the current page, as well as the page in the guidelines on https://www.w3.org/WAI/tutorials/forms/grouping/

The Mac screen reader only reads the checkbox group once the first time you interact with it.  Subsequently, it does not include the group information when reading each checkbox.  This behavior is happening on the Dispatch page, as well as the W3 page.

The code for the page does have the recommended HTML code to group the checkboxes, using a fieldset with a legend (here's a snippet just from the top, and the checkboxes are contained within the fieldset).

```
          <fieldset className="fieldset">
            <legend>
              <label>
                <b>Select Special Issues</b>
              </label>
            </legend>
```

**Comment by leikkisa** [comment]
Here's a screenshot of the first time the voiceover reader interacts with a checkbox within the group. Same happens if it's another checkbox, but only if it's the first one we interact with.

![Screen Shot 2020-06-15 at 3.42.18 PM.png](https://images.zenhubusercontent.com/58f6b6eb51aa2c30577bb60e/779b1c29-16f9-4e4d-902f-e9e793c7a6bf)

**Comment by araposo-tistatech** [comment]
@chandracarney please see the above and advise how we should move forward.

**Comment by leikkisa** [comment]
Checked this out with JAWS today too, and am seeing similar behavior. The first checkbox item gets the group title announced, but not every checkbox. Same for the W3 page.  If navigating away from the checkbox group, and then back again, it reads the group name again for the first checkbox item.

Here's some notes from the JAWS voice text:
**This is from the element above:**
Decision type read only edit 
Full Grant
Use your reading keys to read the text.

**When navigating to the checkbox group - reads group for first one only**
Select Special Issues group 
contaminatedWaterAtCampLejeune check box not checked 
To check press Spacebar.
dicDeathOrAccruedBenefitsUnitedStates check box not checked 
To check press Spacebar.
educationGiBillDependentsEducationalAssistanceScholarshipTransferOfEntitlement check box not checked 
To check press Spacebar.
dicDeathOrAccruedBenefitsUnitedStates check box not checked 
To check press Spacebar.
contaminatedWaterAtCampLejeune check box not checked 
To check press Spacebar.

**Navigating back to the above element**
Decision type read only edit 
Full Grant
Use your reading keys to read the text.

**Reads the group again when we go back**
Select Special Issues group 
contaminatedWaterAtCampLejeune check box not checked 
To check press Spacebar.
